workspace.sh

```diff
@@ -0,0 +1,100 @@
+#!/bin/sh
+#
+#copyright agloo in 2016 under the WTFPL 
+#constructs and groups windows into workspaces to switch between using your favorite keybinder.
+
+NUM_WS=9
+
+help() {
+	cat << EOF
+usage: $(basename $0) [-hinp] [-g ws_num] [-m ws_num] 
+	-h: Displays this message
+	-i: Initializes workspaces and puts you in workspace 0. Should be called once in a startup script.
+	-n: Move up one workspace
+	-p: Move down one workspace
+	-g, --ws_num: go to the workspace specified by ws_num
+	-m, --ws_num: move the currently focused window to the worskpace specified by ws_num
+EOF
+	exit 1
+}
+#initializes us in workspace 0
+ws_init() {
+	mkdir /tmp/workspaces/
+	for i in {0..$NUM_WS}
```

Comment:
`{0..X}` is a bashism, while you shebang reads "`#!/bin/sh`". One of the stricly POSIX solutions to this is as follows:

    i=0
    while [ "$i" -le "$NUM_WS" ]; do
        # something
        i=$(expr $i + 1)
    done